pcsx2/Mdec.cpp

```diff
@@ -176,52 +199,63 @@ void mdecWrite1(u32 data) {
 }
 
 u32 mdecRead0(void) {
-	CDR_LOG("mdec0 read %lx", mdec.command);
+	MDEC_LOG("mdec0 read %lx", mdec.command);
 
 	return mdec.command;
 }
 
 u32 mdecRead1(void) {
-#ifdef CDR_LOG
-	CDR_LOG("mdec1 read %lx", mdec.status);
-#endif
+	MDEC_LOG("mdec1 read %lx", mdec.status);
+
 	return mdec.status;
 }
 
 void psxDma0(u32 adr, u32 bcr, u32 chcr) {
 	int cmd = mdec.command;
 	int size;
 
-	CDR_LOG("DMA0 %lx %lx %lx", adr, bcr, chcr);
+	MDEC_LOG("DMA0 %lx %lx %lx", adr, bcr, chcr);
+
+	if (chcr != 0x01000201) return;
 
-	if (chcr!=0x01000201) return;
+	size = (bcr >> 16)*(bcr & 0xffff);
+	int i;
+	for (i = 0; i<(size); i++) {
+		*(u32*)PSXM(((i + 0) * 4)) = iopMemRead32(adr + ((i + 0) * 4));
+		if (i <20)
+			MDEC_LOG(" data %08X  %08X ", iopMemRead32((adr & 0x00FFFFFF) + (i * 4)), *(u32*)PSXM((i * 4)));
+	}
 
-	size = (bcr>>16)*(bcr&0xffff);
 
-	if (cmd==0x40000001) {
-		u8 *p = (u8*)PSXM(adr);
-		iqtab_init(iq_y,p);
-		iqtab_init(iq_uv,p+64);
+	if (cmd == 0x40000001) {
+		u8 *p = (u8*)PSXM(0); //u8 *p = (u8*)PSXM(adr);
+		iqtab_init(iq_y, p);
+		iqtab_init(iq_uv, p + 64);
 	}
-	else if ((cmd&0xf5ff0000)==0x30000000) {
-		mdec.rl = (u16*)PSXM(adr);
+	else if ((cmd & 0xf5ff0000) == 0x30000000) {
+		mdec.rl = (u16*)PSXM(0); //mdec.rl = (u16*)PSXM(adr);
 	}
 
 	HW_DMA0_CHCR &= ~0x01000000;
 	psxDmaInterrupt(0);
 }
 
+u32 mdecArr2[0x100000] = { 0 };
```

Comment:
move mdecArr2 to the top of the file. Otherwise, you could miss it in the middle of the function